### PR TITLE
EVG-14151 add setup group logging

### PR DESCRIPTION
**Pull request by ablack12**
It's _possible_ that BF-20000 could be fixed by the recent changes to setup group logic, but I think those changes made setup group _less_ strict, not more strict. I'd like to add logging to figure out why we're deciding to run setup group, when another task in the task group had already run immediately before.

### CODE REVIEW

File: public/static/partials/hostevent.html [hunk 1]
```diff
@@ -52,8 +52,8 @@
         </div>
     </span>
     <span ng-switch-when="HOST_PROVISIONED">Marked as <b>provisioned</b></span>
-    <span ng-switch-when="HOST_RUNNING_TASK_SET">Assigned to run task <a href="/task/[[eventLogObj.data.task_id]]/[[eventLogObj.data.execution]]">[[eventLogObj.data.task_id | shortenString:false:50:' ...']]</a></span>
-    <span ng-switch-when="HOST_RUNNING_TASK_CLEARED">Current running task cleared (was: <a href="/task/[[eventLogObj.data.task_id]]/[[eventLogObj.data.execution]]">[[eventLogObj.data.task_id | shortenString:false:50:' ...']]</a></span>
+    <span ng-switch-when="HOST_RUNNING_TASK_SET">Assigned to run task <a href="/task/[[eventLogObj.data.task_id]]/[[eventLogObj.data.execution]]">[[eventLogObj.data.task_id | shortenString:false:100:' ...']]</a></span>
```
Comment (ablack12) [inline public/static/partials/hostevent.html:55]: This is unrelated, I'm just really tired of this page being useless.

File: agent/agent.go [hunk 2]
```diff
@@ -316,10 +320,14 @@ func shouldRunSetupGroup(nextTask *apimodels.NextTaskResponse, tc *taskContext)
 				"next_task_task_group":    nextTask.TaskGroup,
 			})
 		}
-		return true
+		msg = "running setup group because new task group"
+		setupGroup = true
 	}
 
-	return false
+	if tc.logger != nil {
```
Comment (john-m-liu) [inline agent/agent.go:327]: depending on how actively you want to monitor this, you can also log to the agent's log file if the task logger isn't setup

File: agent/agent.go [hunk 3]
```diff
@@ -295,15 +295,19 @@ func (a *Agent) prepareNextTask(ctx context.Context, nextTask *apimodels.NextTas
 
 func shouldRunSetupGroup(nextTask *apimodels.NextTaskResponse, tc *taskContext) bool {
 	// we didn't run setup group yet
+	setupGroup := false
+	var msg string
 	if !tc.ranSetupGroup {
-		return true
+		msg = "running setup group because we haven't yet"
```
Comment (john-m-liu) [inline agent/agent.go:301]: are you intentionally replacing the message each time?

File: public/static/partials/hostevent.html (hunk 1 above)
Comment (john-m-liu) [reply public/static/partials/hostevent.html:55]: the page has feelings ya know 😭

File: agent/agent.go (hunk 3 above)
Comment (ablack12) [reply agent/agent.go:300]: no 😭

**Review by john-m-liu — APPROVED** [review]

**Comment by ablack12** [comment]
evergreen merge
